cmd/server-startup-msg.go

```diff
@@ -67,6 +71,34 @@ func printStartupMessage(apiEndPoints []string) {
 	}
 }
 
+// strip api endpoints list with standard ports such as
+// port "80" and "443" before displaying on the startup
+// banner.  Returns a new list of API endpoints.
+func stripStandardPorts(apiEndpoints []string) (newAPIEndpoints []string) {
+	newAPIEndpoints = make([]string, len(apiEndpoints))
+	// Check all API endpoints for standard ports and strip them.
+	for i, apiEndpoint := range apiEndpoints {
+		url, err := url.Parse(apiEndpoint)
+		if err != nil {
+			newAPIEndpoints[i] = apiEndpoint
+			continue
+		}
+		host, port, _ := net.SplitHostPort(url.Host)
+		// For standard HTTP(s) ports such as "80" and "443"
+		// apiEndpoints should only be host without port.
+		switch {
+		case url.Scheme == "http" && port == "80":
+			fallthrough
+		case url.Scheme == "https" && port == "443":
+			url.Host = host
+			newAPIEndpoints[i] = url.String()
```

Comment:
This means its possible to have valid endpoints `https://server:80` and `http://server:443`.  If TLS is configured in minio it works for both secure/insecure mode where insecure redirects to secure port  ie what is the behavior for `https://server:80` is accessed by `http://server:80`?